### PR TITLE
Blob gas info in receipts

### PR DESCRIPTION
**Pull request by deffrian**
Fixes Closes Resolves #
https://github.com/ethereum/execution-apis/pull/398

## Changes

- DataGasUsed and DataGasPrice

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.JsonRpc/Modules/Parity/ParityRpcModule.cs [hunk 1]
```diff
@@ -77,7 +78,20 @@ public ResultWrapper<ReceiptForRpc[]> parity_getBlockReceipts(BlockParameter blo
...
-                    new ReceiptForRpc(t.Hash, r, t.CalculateEffectiveGasPrice(isEip1559Enabled, block.BaseFeePerGas), receipts.GetBlockLogFirstIndex(r.Index)));
+                {
+                    UInt256 effectiveGasPrice = t.CalculateEffectiveGasPrice(isEip1559Enabled, block.BaseFeePerGas);
+                    UInt256? dataGasPrice = null;
+                    ulong? dataGasUsed = null;
+                    if (t.SupportsBlobs)
+                    {
+                        dataGasPrice = IntrinsicGasCalculator.CalculateDataGasPrice(block.Header, t);
+                        dataGasUsed = IntrinsicGasCalculator.CalculateDataGas(t);
+                    }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Parity/ParityRpcModule.cs:89]: This code seems to get duplicated. Maybe lets calculate `TxGasInfo` in one call?

File: src/Nethermind/Nethermind.Facade/IBlockchainBridge.cs [hunk 2]
```diff
@@ -45,4 +45,23 @@ public interface IBlockchainBridge : ILogFinder
         void RunTreeVisitor(ITreeVisitor treeVisitor, Keccak stateRoot);
 
     }
+
+    public class TxGasInfo
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Facade/IBlockchainBridge.cs:49]: struct

File: src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthRpcModuleTests.cs [hunk 3]
```diff
@@ -840,15 +841,21 @@ public async Task Eth_get_transaction_receipt()
             .WithLogs(entries).TestObject;
         TxReceipt[] receiptsTab = { receipt };
 
-        blockchainBridge.GetReceiptAndEffectiveGasPrice(Arg.Any<Keccak>()).Returns((receipt, UInt256.One, 0));
+        if (postEip4844)
+            blockchainBridge.GetReceiptAndGasInfo(Arg.Any<Keccak>()).Returns((receipt, new(UInt256.One, 2, 3), 0));
+        else
+            blockchainBridge.GetReceiptAndGasInfo(Arg.Any<Keccak>()).Returns((receipt, new(UInt256.One), 0));
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthRpcModuleTests.cs:847]: would prefer `blockchainBridge.GetReceiptAndGasInfo(Arg.Any<Keccak>()).Returns((receipt, postEip4844 ? new(UInt256.One, 2, 3) : new(UInt256.One), 0));`

File: src/Nethermind/Nethermind.JsonRpc/Modules/Parity/ParityRpcModule.cs (hunk 1 above)
Comment (deffrian) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Parity/ParityRpcModule.cs:89]: You mean something like `tx.GetGasInfo()`?

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**